src/react/components/pages/editorPage/canvas.tsx

```diff
@@ -96,10 +141,41 @@ export default class Canvas extends React.Component<ICanvasProps, ICanvasState>
      */
     public onTagClicked = (tag: ITag) => {
         for (const region of this.state.selectedRegions) {
-            this.toggleTagOnRegion(region, tag);
+            CanvasHelpers.toggleTag(region.tags, tag);
+            this.editor.RM.updateTagsById(region.id, CanvasHelpers.getTagsDescriptor(region));
         }
     }
 
+    public onTagShiftClicked = (tag: ITag) => {
+        console.log(`Shift clicked "${tag.name}" tag`);
+    }
+
+    public onTagCtrlShiftClicked = (tag: ITag) => {
+        console.log(`Ctrl shift clicked "${tag.name}" tag`);
+    }
+
+    /**
+     * Method called when deleting a region from the editor
+     * @param {string} id the id of the deleted region
+     * @param {boolean} multiselection boolean whether multiselect is active
+     * @returns {void}
+     */
+    public onRegionSelected = (id: string) => {
```

Comment:
Can this be private?

Reply:
This was previously public, and I think it's to accommodate for [testing ](https://github.com/Microsoft/VoTT/blob/v2/src/react/components/pages/editorPage/canvas.test.tsx#L175)

Reply:
In mocha you can inherit your test class to extend the class you want to test so you have access to private/protected method.  Is there anything similar in jest ?

Also Docs mentioned method to "delete" vs "select"